lib/spack/spack/package.py

```diff
@@ -880,6 +896,11 @@ def do_install(self,
         def build_process():
             """Forked for each build. Has its own process and python
                module space set up by build_environment.fork()."""
+
+            # Set up a global license if one is required
+            if self.license_required:
+                self.set_up_license()
```

Comment:
could this be implemented as a pre-install hook just to keep the logic separate?